desk/app/activity.hoon

```diff
@@ -424,21 +491,58 @@
     ==
   --
 ::
+++  strip-threads
+  |=  =activity:a
+  %-  ~(rep by activity)
+  |=  [[=source:a as=activity-summary:a] out=activity:a]
+  ?:  ?=(?(%thread %dm-thread) -.source)  out
+  (~(put by out) source as)
 ++  base
   ^-  index:a
   (~(got by indices) [%base ~])
 ++  get-index
   |=  =source:a
   (~(gut by indices) source *index:a)
 ++  give-update
-  |=  [=update:a path=(unit path)]
+  |=  $:  =update:a
+        $=  dist
+        $%  [%hose ~]
+            [%both =path]
+            [%only =path]
+        ==
+      ==
   ^+  cor
-  =/  v0-paths  ?~(path ~[/ /v0] ~[/ /v0 u.path [%v0 u.path]])
-  =/  v0-cage=cage  activity-update+!>((update-0:convert-to update))
-  =/  v1-paths  ?~(path ~[/v1] ~[/v1 [%v1 u.path]])
-  =/  v1-cage=cage  activity-update-1+!>(update)
-  =.  cor  (give %fact v1-paths v1-cage)
-  (give %fact v0-paths v0-cage)
+  ~?  verbose  [update dist]
+  =?  cor  ?!(?=(%activity -.update))
+    =?  dist  ?=(%read -.update)  [%both /unreads]
+    =/  v0-paths
+      ?-  -.dist
+        %hose  ~[/ /v0 /v2]
+        %only  ~[path.dist [%v0 path.dist] [%v2 path.dist]]
+        %both  ~[/ /v0 /v2 path.dist [%v0 path.dist] [%v2 path.dist]]
+      ==
```

Comment:
For these and the others, to prevent flubs around path changes/inclusion etc, might be worth it to write the "both" case as being the `+weld` of `%hose` and `%only`. Probably not worth doing a proper loop, but something like the following would be fine.

```suggestion
    =/  v0-paths
      =*  hose  ~[/ /v0 /v2]
      =*  only  ~[path.dist [%v0 path.dist] [%v2 path.dist]]
      ?-  -.dist
        %hose  hose
        %only  only
        %both  (weld hose only)
      ==
```

Reply:
(Doing the version prefixes based on the kind of update being emitted here is a neat approach.)

Reply:
Had to add a face to the `hose` case to do this, and weld wanted more type information, but I got it lol